build/tools/builder-pre.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# NOTE: This script can be run manually but is run anyway by builder.sh before that script starts executing its `main`
+
+# This script will automate the file alteration, which involves
+# 1. Updating the README.md
+# 2. Commiting the changes and pushing the changes
+
```

Comment:
We might want to add PS4 for this script to see the debugging:
```suggestion
export PS4='\e[36m+ ${FUNCNAME:-main}@${BASH_SOURCE}:${LINENO} \e[0m'
```

Reply:
Yes, will add this.